crypto/bio/bss_dgram_pair.c

```diff
@@ -223,6 +292,24 @@ static int dgram_pair_init(BIO *bio)
     return 1;
 }
 
+static int dgram_mem_init(BIO *bio)
+{
+    struct bio_dgram_pair_st *b;
+
+    if (!dgram_pair_init(bio))
+        return 0;
+
+    b = bio->ptr;
+
+    if (ring_buf_init(&b->rbuf, b->req_buf_len) == 0) {
+        ERR_raise(ERR_LIB_BIO, ERR_R_BIO_LIB);
+        return 0;
+    }
+
+    bio->init = 1;
```

Comment:
I'm wondering why this doesn't get set for a dgram_pair in dgram_pair_init?

Reply:
Because the dgram_pair to be in initialized state needs the pair to be paired. Also dgram_pair_init does not actually allocate the ring_buf.